src/platform/datapath_raw_socket.c

```diff
@@ -625,6 +631,7 @@ CxPlatDpRawParseIPv4(
             return;
         }
 
+        Packet->TypeOfService = IP->EcnField;
```

Comment:
FYI, the new ECN code was verified by collecting pcap during the DataPathTest/DataPathTest.UdpDataECT0/4 and DataPathTest/DataPathTest.UdpDataECT0/6 testcases and checking that the packets have ECT0 (instead of, say, some other incorrect ECN bit) set.